tests/index.js

```diff
@@ -1,25 +1,134 @@
-var Repo = require('../repo')
 var Dat = require('..')
 var test = require('tape')
 
+var DAT_PROTOCOL = 'dat://'
+var DAT_KEY_STRING_LENGTH = 64
+var DAT_KEY_BYTE_LENGTH = 32
+var DAT_URL_LENGTH = DAT_PROTOCOL.length + DAT_KEY_STRING_LENGTH
+
 test('create a dat in memory', function (t) {
-  t.plan(7)
+  t.plan(6)
   var dat = new Dat()
   t.equals(dat.repos.length, 0, 'has zero repos before adding')
-  dat.add(function (repo) {
-    t.equals(dat.repos.length, 1, 'has one repo after adding')
-    t.ok(repo.key, 'repo ready and has a key')
-    t.equals(repo.key.length, 32, 'has key with proper length')
-    t.equals(repo.archive.key, repo.key, 'key is the archive key')
+  var repo = dat.add()
+
+  dat.on('repo', (repo) => {
+    t.ok(repo, 'emits the repo event')
+    t.equals(repo.url.length, DAT_URL_LENGTH, 'repo key is there')
+  })
+
+  t.equals(dat.repos.length, 1, 'has one repo after adding')
+  repo.ready(() => {
+    t.equals(repo.archive.key.length, DAT_KEY_BYTE_LENGTH, 'has key with proper length')
+    t.equals(repo.archive.key.toString('hex'), repo.url.slice(DAT_PROTOCOL.length), 'key is the archive key')
+
+    repo.close()
+
+    // TODO: What's this for?
```

Comment:
UI developers might want to respond to some close events by showing the user some feedback, similarly on `error` events

Reply:
Whoops, that comment was there because the callback was empty. It's since become functional because I do `dat.close()` inside it.